### ISSUE TITLE
OpenIdConnectMiddleware: Claims from UserInfo endpoint are not properly mapped to the ClaimsPrincipal

### ISSUE THREAD

**Issue by mclark1129** [aspnet/Security#1037]
I am trying to build my ClaimsPrincipal by using profile claims from the UserInfo endpoint (`GetClaimsFromUserInfoEndpoint = true`), rather than providing them in the `id_token`.  For some reason, it seems like when you have a collection of claims that are the same type these get parsed out differently when they are provided in a JWT vs a simple JSON object from the UserInfo endpoint.  This causes a problem where `User.IsInRole("rolename")` stops working and I cannot Authorize to my site.

The profile claims in my JWT look like this:

```
  // .. All the other JWT Header stuff.. 
  "given_name": "Michael",
  "family_name": "Clark",
  "name": "Michael Clark",
  "role": [
    "Admin",
    "IDENTITY_SERVER_ADMIN"
  ],
  // .. Rest of JWT
```
When the `ClaimsPrincipal` is built from the JWT, I get a single `Claim` of type `role` for every role in the collection.

The claims from the user profile endpoint are just a standard JSON object:
```
{
"given_name":"Michael",
"family_name":"Clark",
"name":"Michael Clark",
"role":[
    "Admin",
    "IDENTITY_SERVER_ADMIN"
],
"sub":"1"
}
```
In this case my `ClaimsPrincipal` ends up with a single claim for `role` with the value of the entire collection of roles.  Now `User.IsInRole("Admin")` will no longer return true.

Looks like the problem originates in `OpenIdConnectHandler.cs:706`

```
if (contentType.MediaType.Equals("application/json", StringComparison.OrdinalIgnoreCase))
{
    user = JObject.Parse(userInfoResponse);
}
```

**Comment by brockallen** [comment]
perhaps dup: https://github.com/aspnet/Security/issues/976

**Comment by mclark1129** [comment]
@brockallen I think you're right, seems like the same issue.  I think that #976 was closed as being fixed, so if thats true any idea when the fix might get released?

**Comment by Tratcher** [comment]
@mclark1129 it was fixed in 1.1.0-preview1 which is already available on nuget.org. The final release should be available shortly.

**Comment by mclark1129** [comment]
@Tracher sounds good, thanks!
